docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -0,0 +1,416 @@
+    <td>
+      <li>Accept scheduling requests of VM/Container submitted from the Scheduling Request Distributor</li>
+      <li>Run the scheduling algorithm to select the best cluster</li>
+      <li>Send the VM/Container Request to the selected cluster which will schedule/select a best node to run the VM/Container and provision related network and storage resources</li>
+      <li>Update Data Store with the scheduling result</li>
+      <li>Update the selected cluster's allocable capacity in the Data Store</li>
+      <li>Internally maintain and manage short run queues of VMs/Containers to be scheduled as well as short run queues of scheduling decision</li>
+      <li>Manage re-scheduling in case of scheduling failure response from the cluster scheduler</li>
+      <li>Support multiple schedulers running concurrently to meet the scalability and latency requirement</li>
+    </td>
+  </tr>
+    
+  <tr>
+    <td>Cluster Resource Collector</td>
+    <td>Collect and Construct the cluster level resource information every few seconds and sends them to the Data Store which will cache the Info</td>
+    <td>
+      <li>Collect allocable capacity information from each cluster in a timely manner</li>
+      <li>Handle the interaction with the underlying Cluster Platform, such as Openstack platform, K8S platform, Arktos platform</li>
+      <li>Support multiple partitions with each partition handling a group of clusters</li>
+      <li>Resource Data Structure design must meet the high query/update frequency and the low latency requirement</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>Flow Monitor</td>
+    <td>Some VM may host applications that involve a lot of interactions with the end-users. Fast response and high throughput are critical requirements for such applications. The VM that hosts such an application needs to be placed near the application's end-user traffic. This Flow Monitor monitors each VM's application flow volume, derive the application flow's origins. It then analyzes the information and determines whether there is a need to migrate the VM to another location or to scale out more VMs to support the application flow.</td>
+    <td>
+      <li>Monitor each VM's application flow volume</li>
+      <li>Retrieve the application flow's origins based on their source IPs</li>
+      <li>Analyzes the application flow's geolocation and volume information to determine whether there is a need to migrate the VM to another location or to scale out more VMs to support the application flow</li>
+      <li>Determine the number of new VMs that need to be created to serve the application traffic flow</li>
+      <li>Send migration instruction to the Migration Manager for VM migration or send scale-out/in instruction to the VM Horizontal Auto Scaler for auto-scaling out/in</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>Migration Manager</td>
```

Comment:
Migration:This interaction need cluster scheduler have the capability to accept scheduler request but not do the create VM task.